### ISSUE TITLE
Problem using save() with createGraphics

### ISSUE THREAD

**Issue by EmmanuelPil** [Calsign/APDE#34]
Hi. First I want to thank you for this excellent App.
It makes coding on Android a lot easier.
The only problem I had until now is with the  code below.
It saves the image but it is a corrupt one.
Using same code on Processing PC Android mode saves the file correctly.
The differences in the two generated files are: 
A is the broken one. Seems to me header is missing.

```
-       ***  Comparing A:	/storage/emulated/0/imgFolder/tempimg.png 	   * regular file * 0 bytes * last status change: 2018-02-21.08:08:07 * last file modification: 2018-02-21.08:08:07 * mode [octal]: 100770     
+       ***  With B:   		/storage/emulated/0/DCIM/Camera/tempimg.png 	   * regular file * 106 bytes * last status change: 2018-02-21.08:07:01 * last file modification: 2018-02-21.08:07:01 * mode [octal]: 100770     
+000000  89 50 4e 47 0d 0a 1a 0a 00 00 00 0d 49 48 44 52 PNG        IHDR     
+000010  00 00 00 0a 00 00 00 0a 08 06 00 00 00 8d 32 cf              2     
+000020  bd 00 00 00 04 73 42 49 54 08 08 08 08 7c 08 64     sBIT    | d     
+000030  88 00 00 00 21 49 44 41 54 18 95 63 64 60 60 f8    !IDAT cd``     
+000040  cf 40 04 60 61 60 60 60 a8 af af 27 a8 90 89 18 @ `a```'     
+000050  d3 46 15 12 04 00 c5 56 02 92 03 5c 06 ad 00 00 F    V  \      
+000060  00 00 49 45 4e 44 ae 42 60 82                     IENDB`     
```

CODE:

```
import android.os.Environment;

PGraphics img;
String s = "tempimg.png";

void setup() {
 // fullScreen();
 
  size(200,200);
  background(255);
  img = createGraphics(10, 10);
  drawImg();
  saveImage(s);
}

void draw() {
  image(img, width/2, height/2);
}
 
  void drawImg() {
  img.beginDraw(); 
  img.fill(127);
  img.rect(0, 0, img.width, img.height);
  img.endDraw();
}

void saveImage(String s) {
   try {
      String directory = new String(Environment.getExternalStorageDirectory().getAbsolutePath() + "/imgFolder");
      File imgFolder = new File(directory);
      if (!imgFolder.exists()) {
        boolean success = true;
       success = imgFolder.mkdirs();
      }
      img.save(directory + "/" + s);
      println("File saved successfully.");
       } 
       catch (Exception e) 
       {
       println("Error while saving file: " + e);
       }
}
```

**Comment by EmmanuelPil** [comment]
My temporary solution for this, is to declare a PImage and draw the img to the screen and then 'get()' it and save it with the saveImage function above.

PImage img;
String s = "tempimg.png";

void setup() {
  size(200,200);
  background(255);
  rect(width/2,height/2,10,10);
  img = get(width/2,height/2,10,10);
  saveImage(s);
}

**Comment by EmmanuelPil** [comment]
Well, like I said above "Using same code on Processing PC Android mode saves the file correctly."

**Comment by Calsign** [comment]
@EmmanuelPil This should be fixed in v0.5.0-pre3 on the [preview channel](https://plus.google.com/communities/117703520591262292626). The change was upgrading to Android mode 4 because there was a bug in the older version. APDE previously used an old version of Android mode, that is why it worked for on the desktop but not in APDE.
